pkg/api/validate/dynamic/quota.go

```diff
@@ -11,114 +11,15 @@ import (
 	"github.com/Azure/ARO-RP/pkg/api"
 )
 
-func addRequiredResources(requiredResources map[string]int, vmSize api.VMSize, count int) error {
+func addRequiredResources(requiredResources map[string]int, vm api.AzureVM, count int) error {
+	if vm.Type == "" || vm.CoreCount < 1 || vm.Family == "" {
+		return fmt.Errorf("unexpected node VMSize %s", vm.Type)
```

Comment:
I believe at this point we should already know that VM size is already valid so there is no need in this check.

Instead we can create an unit test to check that `vmTypesMap` contains all the supported VM sizes to make sure that we do not forget to udpate it when we add new VM sizes.